src/protocols/secure_channel/SessionEstablishmentExchangeDispatch.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines Application Channel class. The object of this
+ *      class can be used by CHIP data model cluster applications to send
+ *      and receive messages. The messages are encrypted using session keys.
+ */
+
+#pragma once
+
+#include <messaging/ExchangeMessageDispatch.h>
+#include <transport/TransportMgr.h>
+
+namespace chip {
+namespace Transport {
+class BLE;
+}
+
+class SessionEstablishmentExchangeDispatch : public Messaging::ExchangeMessageDispatch
+{
+public:
+    SessionEstablishmentExchangeDispatch() {}
+
+    virtual ~SessionEstablishmentExchangeDispatch() {}
+
+    CHIP_ERROR Init(Transport::BLE * transport, TransportMgrBase * transportMgr)
+    {
+        ReturnErrorCodeIf(transport == nullptr && transportMgr == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        mBLETransport = transport;
+        mTransportMgr = transportMgr;
```

Comment:
SessionEstablishmentExchangeDispatch can only use TCP, right? should we add assert check here?

Reply:
Currently, it's only used for PASE, and is using either BLE, or UDP.
I think, we need to define a new delegate for CASE (since BLE is not a valid transport for it).

Reply:
Acked